src/stdune/temp.ml

```diff
@@ -14,3 +14,18 @@ let create ~prefix ~suffix =
 let destroy fn =
   Path.unlink_no_err fn;
   tmp_files := Path.Set.remove !tmp_files fn
+
```

Comment:
We need to:

- have a single `at_exit` for all files and dirs
- import the `Filename.temp_file` logic to retry when the creation fails
- use the same logic for both files and dirs, to keep this code uniform